### ISSUE TITLE
False Positive E1111 assignment-from-no-return for method returning tuple

### ISSUE THREAD

**Issue by jamesquilty** [pylint-dev/pylint#10873]
### Bug description

I'm receiving E1111 assignment-from-no-return message on every occurrence of `a, b = object.method()` inside a module `src/main.py` of this form:

```python
from subdir import module as module_subdir

object = module_subdir.someClass()
...
def somefunc():
    ...
    a, b = object.method()
```

I receive E1111 even when I make the method as simple as possible inside the class:

```python
class someClass:

    def method(self):

        c = -1
        d = -1

        return (c, d)
```

I don't receive E111 for other methods of this class which return multiple values.

### Command used

I invoke pylint with

```shell
pylint src
```

where `src` is the directory tree containing a number of modules including `src/main.py` and `src/subdir/module.py`.

### Pylint output

```text
************* Module main
src/main.py:194:16: E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
src/main.py:194:16: E0633: Attempting to unpack a non-sequence (unpacking-non-sequence)
...
src/main.py:662:16: E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
src/main.py:665:16: E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
src/main.py:670:12: E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
```

### Expected behavior

I'd expect that these errors wouldn't be emitted.

### Pylint version

```text
pylint 4.0.5
astroid 4.0.4
Python 3.11.9 (main, May 11 2024, 12:27:00) [Clang 15.0.0 (clang-1500.3.9.4)]
```

### OS / Environment

macOS 15.7.4

### Additional dependencies

```python

```

**Comment by jacobtylerwalls** [comment]
Thanks, but without a reproducer we can't advance this. At a glance I'm guessing there are namespace package issues frustrating the inference of someClass.
